subsys/fs/littlefs_fs.c

```diff
@@ -181,13 +181,50 @@ static void release_file_data(struct fs_file_t *fp)
 	fp->filep = NULL;
 }
 
-static int littlefs_open(struct fs_file_t *fp, const char *path)
+static int lfs_flags_from_zephyr(int zflags)
+{
+	int flags = 0;
+
+	if (zflags & FO_CREATE) {
+		flags |= LFS_O_CREAT;
+		zflags ^= FO_CREATE;
+	}
+
+	/* LFS_O_READONLY and LFS_O_WRONLY can be selected at the same time,
+	 * this is not a mistake, together they create RDWR access.
+	 */
+	if (zflags & FO_READ) {
+		flags |= LFS_O_RDONLY;
+		zflags ^= FO_READ;
+	}
+
+	if (zflags & FO_WRITE) {
+		flags |= LFS_O_WRONLY;
+		zflags ^= FO_WRITE;
+	}
+
+	if (zflags & FO_APPEND) {
+		flags |= LFS_O_APPEND;
+		zflags ^= FO_APPEND;
+	}
+
+	if (zflags != 0) {
+		/* Only warn, other flags do not affect execution */
+		LOG_WRN("Unsupported open flags: 0x%x", zflags);
+	}
+
+	return flags;
+}
+
+static int littlefs_open(struct fs_file_t *fp, const char *path, int zflags)
 {
 	struct fs_littlefs *fs = fp->mp->fs_data;
 	struct lfs *lfs = &fs->lfs;
-	int flags = LFS_O_CREAT | LFS_O_RDWR;
+	int flags;
```

Comment:
Ideally this (and `ret`) would be initialized in the variable definition, just like `lfs` is.

Reply:
OK